src/aspire/source/eman.py

```diff
@@ -0,0 +1,255 @@
+                L = particle_size
+
+        logger.info(f"Particle size = {L}x{L}")
+        self._original_resolution = L
+
+        original_n = sum([len(self.mrc2coords[x]) for x in self.mrc2coords])
+
+        # Lastly, exclude particle coordinate boxes that do not fit into the micrograph dimensions
+        for _mrc, coordsList in self.mrc2coords.items():
+            out_of_range = []
+            for i in range(len(coordsList)):
+                coord = coordsList[i]
+                start_x, start_y, size_x, size_y = (
+                    coord[0],
+                    coord[1],
+                    coord[2],
+                    coord[3],
+                )
+                if (
+                    start_x < 0
+                    or start_y < 0
+                    or (start_x + size_x >= self.X)
+                    or (start_y + size_y >= self.Y)
+                ):
+                    out_of_range.append(i)
+            # pop in reverse order to avoid messing up indices
+            for j in reversed(out_of_range):
+                coordsList.pop(j)
+
+        n = sum([len(self.mrc2coords[x]) for x in self.mrc2coords])
+        removed = original_n - n
+        logger.info(
+            f"EmanSource from {data_folder} contains {len(self.mrc2coords)} micrographs, {n} picked particles."
+        )
+        logger.info(
+            f"{removed} particles did not fit into micrograph dimensions at particle size {L}, so were excluded."
+        )
+        ImageSource.__init__(self, L=L, n=n, dtype=dtype)
+
+        # Create filter indices for the source. These are required in order to pass through filter eval code
+        self.filter_indices = np.zeros(self.n, dtype=int)
+        self.unique_filters = [IdentityFilter()]
+
+    def _images(self, start=0, num=np.inf, indices=None):
+        # the indices passed to this method refer to the index
+        # of the *particle*, not the micrograph
+        if indices is None:
+            indices = np.arange(start, min(start + num, self.n))
+        else:
+            start = indices.min()
+        logger.info(f"Loading {len(indices)} images from micrographs")
+
+        # explode mrc2coords into a flat list
+        all_particles = []
+        # identify each particle as e.g. 000001@mrcfile, analogously to Relion
+        for mrc in self.mrc2coords:
+            for i in range(len(self.mrc2coords[mrc])):
+                all_particles.append(f"{i:06d}@{mrc}")
+        # select the desired particles from this list
+        _particles = [all_particles[i] for i in indices]
+        # initialize empty array to hold particle stack
+        im = np.empty(
+            (len(indices), self._original_resolution, self._original_resolution),
+            dtype=self.dtype,
+        )
+
+        def crop_micrograph(data, coord):
+            start_x, start_y, size_x, size_y = coord[0], coord[1], coord[2], coord[3]
+            # according to MRC 2014 convention, origin represents
```

Comment:
Thanks for documenting where the convention comes from.  That is a small detail that is really helpful to others.